homeassistant/components/backup/agent.py

```diff
@@ -31,27 +37,25 @@ class BackupAgent(abc.ABC):
     async def async_download_backup(
         self,
         backup_id: str,
-        *,
-        path: Path,
         **kwargs: Any,
-    ) -> None:
+    ) -> BackupAgentStream:
         """Download a backup file.
 
         :param backup_id: The ID of the backup that was returned in async_list_backups.
-        :param path: The full file path to download the backup to.
+        :return: Either an aiohttp or asyncio StreamReader.
         """
```

Comment:
We should maybe let this return an `AsyncGenerator` instead, similar to `async_upload_backup`

Reply:
It's annoying to have two different types.

Reply:
Fixed